url/normalize.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+import { posixNormalize } from "../path/_normalize.ts";
+
+/**
+ * Normalize the `url`, resolving mutiple `'/'`s into `'//'` after protocol and remaining into `'/'`.
```

Comment:
I'd expect normalize to normalize away `..` -- Is this the case?

Reply:
Can you provide me with a test case?

Reply:
Something like
```ts
  [
    "https://deno.land/std/assert/../async/retry.ts",
    new URL("https://deno.land/std/async/retry.ts"),
  ],
  ```
  I'd also probably test that `https://deno.land/..` normalizes to `https://deno.land/`

Reply:
Both the cases pass the test.

Reply:
Added the cases.

Reply:
Could you add that it handles this to the JSDoc?

Reply:
Added.